app/lib/infrastructure/language_repository.dart

```diff
@@ -2,132 +2,174 @@ part of 'package:word_link/domain/controllers/language_controller.dart';
 
 class _LanguageRepository extends LanguageController {
   @override
-  CollectionObject getMostUsedWords(
-      {required LanguageEnum language, int numberOfWords = 5,}) {
-    final Map<String, String> wordMap = _getWordMapForLanguage(language);
+  Future<CollectionObject> getMostUsedWords({
+    required LanguageEnum language,
+    int numberOfWords = 5,
+  }) async {
+    final List<String> words = _getSourceLanguageWords(language);
 
-    final wordPairs = wordMap.entries
+    final Map<String, String> wordPairs = await _getTranslatedWords(
+        language.translateLang, words.sublist(0, numberOfWords));
+
+    final cards = wordPairs.entries
         .take(numberOfWords)
         .map((entry) => CardObject(name: entry.key, answer: entry.value))
         .toList();
+
     return CollectionObject(
-        name: LanguageEnum.polish.displayName, cardsTemp: wordPairs,);
+      name: LanguageEnum.polish.displayName,
+      cardsTemp: cards,
+    );
   }
 
-  Map<String, String> _getWordMapForLanguage(LanguageEnum language) {
+  List<String> _getSourceLanguageWords(LanguageEnum language) {
```

Comment:
The class is private, so having private methods does not do anything.
Remove the _ in the name of all of them